core/common/src/main/scala/org/typelevel/otel4s/TextMapUpdater.scala

```diff
@@ -45,3 +47,23 @@ trait TextMapUpdater[A] {
     */
   def updated(carrier: A, key: String, value: String): A
 }
+
+object TextMapUpdater {
+  def apply[A](implicit updater: TextMapUpdater[A]): TextMapUpdater[A] = updater
+
+  implicit def forMap[CC[x, +y] <: MapOps[x, y, CC, CC[x, y]]]
+      : TextMapUpdater[CC[String, String]] =
+    (carrier: CC[String, String], key: String, value: String) =>
+      carrier.updated(key, value)
+
+  implicit def forSortedMap[
```

Comment:
I think I missed why this one was necessary on top of `forMap`.

Reply:
the very concise answer is that fully generic map implementations extend `MapOps[K, V, CC, CC[K, V]]`, but `SortedMapOps[K, V, CC, CC[K, V]] extends MapOps[K, V, Map, CC[K, V]]`